src/main/java/teammates/logic/api/GateKeeper.java

```diff
@@ -327,21 +327,27 @@ public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttrib
             String section, String email) {
         InstructorAttributes instructor = instructorsLogic.getInstructorForGoogleId(courseId, account.googleId);
         if (instructor != null) {
-            if (!instructor.isAllowedForPrivilege(section,
-                    Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
-                throw new UnauthorizedAccessException("Instructor does not have enough privileges to view the photo");
-            }
+            verifyInstructorCanViewPhoto(instructor, section);
             return;
         }
-
         StudentAttributes student = studentsLogic.getStudentForCourseIdAndGoogleId(courseId, account.googleId);
         if (student != null) {
-            if (!studentsLogic.isStudentsInSameTeam(courseId, email, student.email)) {
-                throw new UnauthorizedAccessException("Student does not have enough privileges to view the photo");
-            }
+            verifyStudentCanViewPhoto(student, email, courseId);
             return;
         }
-
         throw new UnauthorizedAccessException("User is not in the course that student belongs to");
     }
+
+    private void verifyInstructorCanViewPhoto(InstructorAttributes instructor, String section) {
+        if (!instructor.isAllowedForPrivilege(section,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
```

Comment:
This can be combined with the previous line as the resulting line still fits the 125-character limit.